In [541]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import json
from stravalib.client import Client

In [561]:
# Prepare data frames
df = pd.DataFrame(columns=['start_date', 'id', 'elapsed_time', 'map'])
df_temp = pd.DataFrame(columns=df.columns)

In [543]:
# Setup access to Strava API
with open('config.json') as data_file:    
    passwords = json.load(data_file)

client = Client()
access_token = passwords['passwords'][0]['access_token']
client_id = passwords['passwords'][1]['client_id']
client_secret = passwords['passwords'][2]['client_secret']

client.access_token = access_token

In [563]:
# Fetch activity data from Strava_API
for activity in client.get_activities():
    df_temp['start_date'] = [activity.start_date_local]
    df_temp['id'] = [activity.id]
    df_temp['elapsed_time'] = [activity.elapsed_time]
    df_temp['map'] = [activity.map]
    df = df.append(df_temp, ignore_index=True)

In [ ]:
# Analize data from Strava 
thresh_upper = np.timedelta64(100,"m")  # Upper bound in minutes
thresh_lower = np.timedelta64(30,"m")  # Lower bound in minutes

mask = df['elapsed_time'][20:].apply(lambda x: 
                                     True if (x > thresh_lower and x < thresh_upper) 
                                     else False)

filtered_array = df['elapsed_time'][20:].where(mask)

filtered_array.mean()

In [ ]:
plt.scatter()

In [475]:
filtered_array.size - np.count_nonzero(np.isnat(filtered_array))

168

#Average commute time using CalTrain is 1:00 using 168 commutes for data

In [540]:
print(df['map'][1].summary_polyline)
print(df['id'][1])

IndexError: index out of bounds